src/core/lib/channel/promise_based_filter.h

```diff
@@ -263,6 +269,92 @@ class BaseCallData : public Activity, private Wakeable {
     return p.release();
   }
 
+  class SendMessage {
```

Comment:
Please add a comment explaining what these two classes are, in particular the fact that the reason they are split into their own classes is so that we don't need to allocate their state for filters that are not intercepting send or recv messages.

Reply:
done